samza-core/src/main/java/org/apache/samza/coordinator/stream/messages/SetChangelogMapping.java

```diff
@@ -58,7 +58,8 @@ public String getTaskName() {
     return getKey();
   }
 
-  public int getPartition() {
-    return Integer.parseInt(getMessageValue(CHANGELOG_VALUE_KEY));
+  public Integer getPartition() {
+    String changelogPartition = getMessageValue(CHANGELOG_VALUE_KEY);
+    return (changelogPartition != null) ? Integer.parseInt(changelogPartition) : null;
```

Comment:
Probably safer to replace the null check with `StringUtils.isNotBlank(changelogPartition)`

Reply:
Done.